torch/csrc/api/include/torch/nn/cursor.h

```diff
@@ -0,0 +1,196 @@
+#pragma once
+
+#include <torch/csrc/autograd/variable.h>
+
+#include <cstddef>
+#include <iterator>
+#include <limits>
+#include <string>
+#include <type_traits>
+
+// Forward declarations.
+namespace torch {
+namespace detail {
+template <typename T>
+struct CursorCollector;
+} // namespace detail
+namespace nn {
+class Module;
+} // namespace nn
+} // namespace torch
+
+namespace torch {
+namespace detail {
+/// A cursor provides hierarchical iteration support, with convenient iterator
+/// functions like `map` or `find`.
+///
+/// Fundamentally, cursors are similar to `std::map`, with `[]`, `.at()`,
+/// `.size()` etc. When iterating over a cursor, use `.key()` to get the
+/// associated string, and dereference the returned item or use `.value()` to
+/// get the associated value (e.g. the variable, the module or the buffer).
+///
+/// Note that cursors *eagerly* collect items. So if you want to perform many
```

Comment:
I don't know what eagerly collecting an item means.

Reply:
all items immediately, as opposed to lazily producing them like a geneartor